### ISSUE TITLE
.Net: BingConnector and all other IWebSearchEngineConnector implmentations need have original URL or original Json 

### ISSUE THREAD

**Issue by waynebaby** [microsoft/semantic-kernel#4449]
For all the customers which need SK search on web, usually need the snippet to check if the record is worthy to look into, and then open the url to confirm they need the content of  the page. event they are using chain tasks usually customers needs the URLs of reference for integrity. 

current implementations are useless, so no one is using  IWebSearchEngineConnector but create a websearch plugin of their own.  at least make "  original json for each result string"   as a possible output in

``` c sharp
public interface IWebSearchEngineConnector
{
 
    Task<IEnumerable<string>> SearchAsync(string query, int count = 1, int offset = 0, CancellationToken cancellationToken = default);  //add here 
}

```

**Comment by dmytrostruk** [comment]
@waynebaby Thank you for feedback! Just want to point out that plugins that are located in repository are for demonstration purposes only and it's not recommended to use them in production environments. If you need more advanced scenarios for web searching, it is recommended to implement your own plugin that will cover your cases and import it into Kernel to combine it with AI. 

Please let me know if that makes sense, thank you!

**Comment by matthewbolanos** [comment]
There are a handeful of plugins that are likely necessary for most projects (I would include web ones in that bucket) that we'll likely need to make v1 (non-experimental) in the medium-term.

**Comment by waynebaby** [comment]
@dmytrostruk 
I understand the context of "demonstration purposes", however, I noticed that it was included with the Microsoft.SemanticKernel.Plugins.Web package (I haven't checked the newer versions, but it was there in my 1.0.0-beta8). It gives off an official impression, and it's a sealed class, which doesn't allow for extensions. My main concern may be "why was it shipped in this manner".

Perhaps, we could consider a slight enhancement by shipping it as an extendable class with the URL included in the text feature, which could potentially result in a more satisfactory outcome for everyone.

And by the way, Happy New Year!

**Comment by SOE-YoungS** [comment]
I believe this is addressed in #5673
